include/spdlog/sinks/qt_sinks.h

```diff
@@ -162,15 +162,24 @@ class qt_color_sink : public base_sink<Mutex>
 
         const string_view_t str = string_view_t(formatted.data(), formatted.size());
         // apply the color to the color range in the formatted message.
-        auto payload = QString::fromLatin1(str.data(), static_cast<int>(str.size()));
+        auto payload = QString::fromUtf8(str.data(), static_cast<int>(str.size()));
+        // convert color ranges from byte index to character index.
+        int color_range_start, color_range_end;
+        color_range_start = static_cast<int>(msg.color_range_start);
+        color_range_end = static_cast<int>(msg.color_range_end);
+        if (msg.color_range_start < msg.color_range_end) {
+            QByteArray bytes(str.data(), str.size());
```

Comment:
No need to create a deep copy of QByteArray. You can use directly
```c++
color_range_start = QString::fromUtf8(str.data(), msg.color_range_start).size();
color_range_end = QString::fromUtf8(str.data(), msg.color_range_end).size();

```

Reply:
Updated.